src/BitIntegers.jl

```diff
@@ -331,6 +331,10 @@ flipsign(x::T, y::T) where {T<:XBS} = flipsign_int(x, y)
 flipsign(x::UBS, y::UBS) = flipsign_int(promote(x, y)...) % typeof(x)
 
 
+# Cheaper isodd, to avoid BigInt.  NOTE: Base.iseven is defined in terms of isodd.
+isodd(a::BitInteger) = isodd(a % Int)  # only depends on the final bit! :)
```

Comment:
This has to be `XBI` instead of `BitInteger`, otherwise this won't work for types defined in user code.